blog/r-tidyfinance-0-4-0-new-data-downloads/index.qmd

```diff
@@ -0,0 +1,126 @@
+
+As the new release brings many new features, we split them up into two blog posts. In this blog post, we discuss the new data downloads that `tidyfinance` now supports. 
+
+## Install the package
+
+You can install the released version of `tidyfinance` from CRAN via:
+
+```{r}
+#| eval: false
+install.packages("tidyfinance")
+```
+
+You can install the development version of tidyfinance from GitHub using the `pak` package:
+
+```{r}
+#| eval: false
+pak::pak("tidy-finance/r-tidyfinance")
+```
+
+You then load the package via:
+
+```{r}
+library(tidyfinance)
+```
+
+## Download stock data
+
+You can download stock prices from [Yahoo Finance](https://finance.yahoo.com/) using the type `stock_prices` and provide symbols:
+
+```{r}
+download_data("stock_prices", symbols = c("AAPL", "MSFT"))
+```
+
+As you can see, we included defaults for `start_date` and `end_date` that you can of course overwrite:
+
+```{r}
+download_data("stock_prices", symbols = c("AAPL", "MSFT"),
+              start_date = "2020-01-01", end_date = "2020-12-31")
+```
+
+To inspect details about the download approach, you can call the documentation via `?download_data_stock_prices`.
+
+To get symbols, you have now two options: first, you can download symbols from the [Nasdaq Stock Screener](https://www.nasdaq.com/market-activity/stocks/screener) for the exchanges AMEX, NASDAQ, and NYSE. Analogously, you can find details using `?download_data_symbols`.
+
+```{r}
+download_data("symbols", exchange = "NASDAQ")
+```
+
+Second, you can  download index constituents from selected [iShares ETFs](https://www.ishares.com/us/products/etf-investments#/?productView=etf&pageNumber=1&sortColumn=totalNetAssets&sortDirection=desc&dataView=keyFacts) that physically replicate the corresponding index:
+
+```{r}
+download_data("constituents", index = "DAX")
+```
+
+The documentation `?download_data_constituents` provides details. 
+
+## Download macro data
+
+We also added support for the [Federal Reserve Economic Databa (FRED)](https://fred.stlouisfed.org/), where you can easily download multiple series (see `?download_data_fred` for details):
+
+```{r}
+download_data("fred", series = c("GDP", "CPIAUCNS"))
+```
+
+Our approach is a simple wrapper around the FRED download data site. If you want to systematically download FRED data via API, please consider using `fredr` package.
+
+The [macro predictors](https://www.tidy-finance.org/r/accessing-and-managing-financial-data.html#macroeconomic-predictors) by Ivo Welch and Amig Goyal now also come in different flavors: monthly, quarterly, and annual. For instance, the annual data can be fetched via:
+
+```{r}
+download_data("macro_predictors_annual")
```

Comment:
This function is actually doing much more! 

``
  processed_data <- processed_data |>
    mutate(
      across(where(is.character), ~ as.numeric(gsub("\\,", "", .))),
      IndexDiv = Index + D12,
      logret = log(IndexDiv) - log(lag(IndexDiv)),
      rp_div = lead(logret - Rfree, 1),
      log_d12 = log(D12),
      log_e12 = log(E12),
      dp = log_d12 - log(Index),
      dy = log_d12 - log(lag(Index)),
      ep = log_e12 - log(Index),
      de = log_d12 - log_e12,
      tms = lty - tbl,
      dfy = BAA - AAA
    ) |>
    select(date, rp_div, dp, dy, ep, de, svar, bm = `b.m`, ntis, tbl, lty, ltr,
           tms, dfy, infl
    ) |>
    tidyr::drop_na()
``
We do not only download the data but prepare it for immediate use. The dataset contains... blabla